src/components/Executions/filters/useUserByFilterState.ts

```diff
@@ -0,0 +1,53 @@
+import { FilterOperationName } from 'models/AdminEntity/types';
+import { useEffect, useState, useContext } from 'react';
+import { FilterByUserState } from './types';
+import { useFilterButtonState } from './useFilterButtonState';
+import { APIContext } from 'components/data/apiContext';
+
+interface FilterByUserStateArgs {
+    queryStateKey: string;
+    filterOperation?: FilterOperationName;
+}
+
+/** Maintains the state for a button to be used for filtering by user.
+ */
+export function useUserByFilterState({
+    queryStateKey,
+    filterOperation = FilterOperationName.EQ
+}: FilterByUserStateArgs): FilterByUserState {
+    const { profile } = useContext(APIContext);
+    const username =
+        profile?.value == null
+            ? ''
+            : profile.value?.preferredUsername == ''
+            ? profile.value.name
+            : profile.value?.preferredUsername;
+
+    const [value, setValue] = useState(username);
+    const button = useFilterButtonState();
+
+    useEffect(() => {
+        if (button.open) setValue('');
+        else setValue(username);
+    }, [button.open]);
```

Comment:
I think we may want to use `active` and `setActive` to track if the user has enabled the filter. And I don't think we need a separate button state.

```ts
// This is tracking whether the user has enabled the filter.
// Depending on what default behavior you want, setting this to true will either 
// show all executions or only show executions belonging to the current user. 
// I think based on the logic below, true === show executions for the current user.
const [active, setActive] = useState(true); 
const getFilter = () => {
        return active && userName
            ? [
                  {
                      value: userName,
                      key: queryStateKey,
                      operation: FilterOperation.EQ
                  }
              ]
            : [];
    };
```